common/flatpak-dir.c

```diff
@@ -5266,7 +5266,7 @@ repo_pull (OstreeRepo                           *self,
            FlatpakRemoteState                   *state,
            const char                          **dirs_to_pull,
            const char                           *ref_to_fetch,
-           const char                           *rev_to_fetch, /* (nullable) */
+           const char                           *rev_to_fetch,
```

Comment:
If this is no longer nullable, a `g_return_val_if_fail (rev_to_fetch != NULL, FALSE)` would be good to have.

Reply:
My understanding is `g_return_val_if_fail`/`g_return_if_fail` is for when it's possible for some calling code to not meet the condition being checked, like if it's library API. And `g_assert` is for when the condition is guaranteed by other code within the library, so I believe an assert would be more appropriate here if we want to add something. On the other hand adding assertions for things that aren't definitely required for correct operation makes the code more brittle, whereas `g_return_val_if_fail` will not cause a crash except in debugging scenarios.

In light of all that, do you still want the `g_return_val_if_fail`?

Reply:
I tend to interpret "possible for some calling code to not meet the condition being checked" fairly broadly, and use `g_return_val_if_fail()` even in internal functions, if it isn't a hot path and it isn't completely obvious that the caller and the callee are so tightly coupled that they will always change in lockstep.

I'm not going to insist, though.

Reply:
Added that check and one for `ref_to_fetch`